packages/react-dom-bindings/src/client/ReactDOMComponent.js

```diff
@@ -2546,10 +2546,10 @@ function diffHydratedCustomComponent(
       // Fall through
       default: {
         let ownNamespaceDev = parentNamespaceDev;
-        if (ownNamespaceDev === HTML_NAMESPACE) {
+        if (ownNamespaceDev === HTML_NAMESPACE || ownNamespaceDev === null) {
```

Comment:
do you still need to check for HTML_NAMESPACE in all these?

Reply:
Yeah I think maybe not